source/translated_table_model.ts

```diff
@@ -28,22 +45,158 @@ export class TranslatedTableModel extends TableModel {
    *                      to this table's columnCount.
    * @throws RangeError - The index specified is not within range.
    */
-  public moveRow(source: number, destination: number): void {}
+  public moveRow(source: number, destination: number): void {
+    if(source >= this.rowCount || source < 0) {
+      throw RangeError();
+    }
+    if(destination >= this.rowCount || destination < 0) {
+      throw RangeError();
+    }
+    this.beginTransaction();
+    const traveller = (() => {
+      if(this.references[source] >= 0) {
```

Comment:
How can this ever be less than 0?

Reply:
When I wrote `if(this.references[source])` leads to a logic error for reasons I haven't been able to figure out. This fixed the issue for reasons I can't fathom.

Reply:
Don't use ```undefined``` for anything. You can use ```null``` and then test for ```null``` with ```x === null```.

Reply:
Actually, it's because if source[index]  is 0 it treats it like a false even though there is a index. I could do `if(this.references[source]  !==  null)` if you would prefer that.